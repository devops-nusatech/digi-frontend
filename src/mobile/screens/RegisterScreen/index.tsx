import * as React from 'react';
import { useIntl } from 'react-intl';
import { useHistory } from 'react-router';
import { RegisterScreen } from '../../../screens/RegisterScreen';
import { Modal } from '../../components';

const RegisterMobileScreen: React.FC = () => {
   const history = useHistory();
   const intl = useIntl();

   return (
      <div>
         <Modal
            isOpen
            onClose={() => history.push('/trading')}
            onBack={() => history.push('/login')}
            backTitle={intl.formatMessage({
               id: 'page.body.landing.header.button2',
            })}
            title={intl.formatMessage({
               id: 'page.body.landing.header.button3',
            })}>
            <RegisterScreen />
         </Modal>
      </div>
   );
};

export { RegisterMobileScreen };
