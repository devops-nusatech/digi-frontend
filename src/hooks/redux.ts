import { shallowEqual, useSelector } from 'react-redux';
import { RootState } from 'modules';

export const useReduxSelector = <T>(selector: (state: RootState) => T): T =>
   useSelector(selector, shallowEqual);
