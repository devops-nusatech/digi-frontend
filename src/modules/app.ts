import { combineReducers } from 'redux';
import { airdropsReducer } from './public/airdrops';
import { alertReducer } from './public/alert';
import { blockchainsReducer } from './public/blockchains';
import { blocklistAccessReducer } from './public/blocklistAccess';
import { configsReducer } from './public/configs';
import { currenciesReducer } from './public/currencies';
import { customizationReducer } from './public/customization';
import { errorHandlerReducer } from './public/errorHandler';
import { changeColorThemeReducer } from './public/globalSettings';
import { gridLayoutReducer } from './public/gridLayout';
import { changeLanguageReducer } from './public/i18n';
import { klineReducer } from './public/kline';
import { marketsReducer } from './public/markets';
import { memberLevelsReducer } from './public/memberLevels';
import { membershipsReducer } from './public/memberships';
import { newsReducer } from './public/news';
import { news2Reducer } from './public/news2';
import { depthReducer, incrementDepthReducer, orderBookReducer } from './public/orderBook';
import { rangerReducer } from './public/ranger/reducer';
import { recentTradesReducer } from './public/recentTrades';
import { tradingFeesReducer } from './public/tradingFees';
import { withdrawLimitsReducer } from './public/withdrawLimits';
import { apiKeysReducer } from './user/apiKeys';
import { authReducer } from './user/auth';
import { beneficiariesReducer } from './user/beneficiaries';
import { getGeetestCaptchaReducer } from './user/captcha';
import { customizationUpdateReducer } from './user/customization';
import { documentationReducer } from './user/documentation';
import { sendEmailVerificationReducer } from './user/emailVerification';
import { groupMemberReducer } from './user/groupMember';
import { historyReducer } from './user/history';
import {
   addressesReducer,
   documentsReducer,
   identityReducer,
   labelReducer,
   phoneReducer,
} from './user/kyc';
import { openOrdersReducer } from './user/openOrders';
import { ordersReducer } from './user/orders';
import { ordersHistoryReducer } from './user/ordersHistory';
import { passwordReducer } from './user/password';
import { profileReducer } from './user/profile';
import { tierReducer } from './user/tier';
import { transactionsReducer } from './user/transactions';
import { transferReducer } from './user/transfers';
import { userActivityReducer } from './user/userActivity';
import { walletsReducer } from './user/wallets';
import { withdrawLimitReducer } from './user/withdrawLimit';

export const publicReducer = combineReducers({
   airdrops: airdropsReducer,
   alerts: alertReducer,
   blockchains: blockchainsReducer,
   blocklistAccess: blocklistAccessReducer,
   colorTheme: changeColorThemeReducer,
   configs: configsReducer,
   currencies: currenciesReducer,
   customization: customizationReducer,
   errorHandler: errorHandlerReducer,
   rgl: gridLayoutReducer,
   i18n: changeLanguageReducer,
   kline: klineReducer,
   markets: marketsReducer,
   memberLevels: memberLevelsReducer,
   memberships: membershipsReducer,
   news: newsReducer,
   news2: news2Reducer,
   orderBook: orderBookReducer,
   depth: depthReducer,
   incrementDepth: incrementDepthReducer,
   ranger: rangerReducer,
   recentTrades: recentTradesReducer,
   tradingFees: tradingFeesReducer,
   withdrawLimits: withdrawLimitsReducer,
});

export const userReducer = combineReducers({
   addresses: addressesReducer,
   apiKeys: apiKeysReducer,
   auth: authReducer,
   beneficiaries: beneficiariesReducer,
   captcha: getGeetestCaptchaReducer,
   customizationUpdate: customizationUpdateReducer,
   documentation: documentationReducer,
   history: historyReducer,
   documents: documentsReducer,
   identity: identityReducer,
   label: labelReducer,
   phone: phoneReducer,
   openOrders: openOrdersReducer,
   orders: ordersReducer,
   ordersHistory: ordersHistoryReducer,
   password: passwordReducer,
   profile: profileReducer,
   tier: tierReducer,
   groupMember: groupMemberReducer,
   transactions: transactionsReducer,
   transfer: transferReducer,
   sendEmailVerification: sendEmailVerificationReducer,
   userActivity: userActivityReducer,
   wallets: walletsReducer,
   withdrawLimit: withdrawLimitReducer,
});
