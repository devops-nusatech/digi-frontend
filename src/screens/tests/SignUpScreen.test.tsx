import { shallow } from 'enzyme';
import * as React from 'react';
import { connect, Provider } from 'react-redux';
import { createStore } from 'redux';
import { IntlProps } from '../../';
import { rootReducer } from '../../modules';
import { RegisterScreen } from '../RegisterScreen';

const store = createStore(rootReducer);
const Register = connect()(RegisterScreen);

const setup = (props: Partial<IntlProps> = {}) =>
   shallow(
      <Provider store={store}>
         <Register />
      </Provider>
   );

describe('RegisterScreen', () => {
   const wrapper = setup();

   it('should render', () => {
      expect(wrapper).toMatchSnapshot();
   });
});
