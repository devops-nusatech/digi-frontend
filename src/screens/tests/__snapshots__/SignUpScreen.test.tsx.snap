// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen should render 1`] = `ShallowWrapper {}`;
